msgp/read.go

```diff
@@ -759,6 +759,45 @@ func (m *Reader) ReadBytes(scratch []byte) (b []byte, err error) {
 	return
 }
 
+// ReadBytesHeader reads the size header
+// of a MessagePack 'bin' object. The user
+// is responsible for dealing with the next
+// 'sz' bytes from the reader in an application-specific
+// way.
+func (m *Reader) ReadBytesHeader() (sz uint32, err error) {
+	var p []byte
+	p, err = m.r.Peek(1)
+	if err != nil {
+		return
+	}
+	switch p[0] {
+	case mbin8:
+		p, err = m.r.Next(2)
```

Comment:
hi @philhofer--that makes sense. My messages are 100 bytes +/- 20 bytes. After an io.Copy the last one of them read in will have been truncated, 99% of the time. For my app its a common need to recover a bin that is ~100 bytes easily and without fuss.

I understand if you don't want to support two different transactional read modes.  In that case I'd appreciate it if you'd merge PR #110 so I don't have to maintain a separate branch to move forward, but even that isn't really a big deal. It's already branched and I've moved on to the next thing, so either way is really fine.